File: 프로그래머스/0/120846. 합성수 찾기/합성수 찾기.py
def solution(n):
    answer = 0                      # 합성수의 개수

    for i in range(1, n + 1):       # n이하의 자연수 이므로 1부터 n까지 반복
        count = 0                   # i의 약수에 대한 카운트 변수

        # 1부터 i까지 반복하면서 약수의 개수를 카운트한다.
        for j in range(1, i+1):
            if i % j == 0:          # j가 i의 약수라면?
                count += 1          # 카운트 1 증가
                if count >= 3:      # 약수의 개수가 3개 이상이라면?
                    answer += 1     # 합성수의 개수 1증가
                    break           # 3개를 초과하는 약수의 수는 의미 없으므로 반복 중단

    return answer

print(solution(10))     # 5
print(solution(15))     # 8

File: 프로그래머스/0/120846. 합성수 찾기/합성수 찾기.go
// 합성수 개수를 구하는 함수
func solution(n int) int {
	answer := 0 // 합성수의 개수를 저장할 변수 초기화
	for i := 1; i <= n; i++ {
		count := 0 // i의 약수 개수를 세는 변수 초기화
		for j := 1; j <= i; j++ {
			if i%j == 0 {
				count++
				if count >= 3 {
					answer++
					break
				}
			}
		}
	}
	return answer // 계산된 합성수 개수 반환
}